### PR TITLE
Trivial improvement on array join 

### PR DESCRIPTION
**Pull request by taiyang-li**
### Changelog category (leave one):
- Performance Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Trivial improvement on array join, reuse some intermediate results.

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 35cf4cd1dde0bc28adf378f98721c9c3d0d37a9e with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/57183/35cf4cd1dde0bc28adf378f98721c9c3d0d37a9e/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6977939037">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/57183/35cf4cd1dde0bc28adf378f98721c9c3d0d37a9e/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by Algunenano** [comment]
Tests are broken

**Comment by alexey-milovidov** [comment]
Need to fix or close.

File: src/Interpreters/ArrayJoinAction.cpp [hunk 1]
```diff
@@ -206,10 +206,19 @@ Block ArrayJoinResultIterator::next()
...
     auto cut_any_col = any_array->cut(current_row, next_row - current_row);
     const auto * cut_any_array = typeid_cast<const ColumnArray *>(cut_any_col.get());
+
     for (size_t i = 0; i < num_columns; ++i)
     {
         ColumnWithTypeAndName current = block.safeGetByPosition(i);
-        current.column = current.column->cut(current_row, next_row - current_row);
+
+        /// Reuse cut_any_col if possible to avoid unnecessary cut.
+        if (!is_unaligned && !is_left && current.name == *columns.begin())
```
[the remark this answers is not in the document]
Comment (taiyang-li) [reply src/Interpreters/ArrayJoinAction.cpp:215]: Yes, it is still correct.

[the remark this answers is not in the document]
Comment (taiyang-li) [reply src/Interpreters/ArrayJoinAction.cpp:215]: As long as any_array is transformed from the first array join column,  comparision by name is right. 
If array join column is huge, then current improvement could avoid cutting twice from array join column

[the remark this answers is not in the document]
Comment (taiyang-li) [reply src/Interpreters/ArrayJoinAction.cpp:215]: @Algunenano I'll fix the issue you mentioned before.

**Review by Algunenano — APPROVED** [review]

**Pull request merged**